### DISCUSSION TITLE
Trouble building stf_tools with bintuils

### DISCUSSION THREAD

**Discussion by knute-mips** [sparcians/stf_tools#3] [General] [answered]
This is more of an issue with binutils I think, but when the step comes to build libintl.a, I get this error:
```
[ 84%] Performing build_libintl step for 'binutils'
plural.c.y:51.1-7: warning: POSIX Yacc does not support %define [-Wyacc]
   51 | %define api.pure full
      | ^~~~~~~
plural.c.y:52.1-7: warning: POSIX Yacc does not support %expect [-Wyacc]
   52 | %expect 7
      | ^~~~~~~
config.status: creating config.h
config.status: config.h is unchanged
plural.c.c:146:10: fatal error: plural.c.h: No such file or directory
```

I upgraded `bison` to 3.8.2, but problem still persists.

The machine is a CentOS box, Rocky Linux release 8.8 (Green Obsidian)

**Comment by bdutro** [comment]
Can you post the contents of `binutils-prefix/src/binutils-build/config.log`?

**Comment by knute-mips** [comment]
BTW, this is broken too.  ☹️ 
```
$ cmake -DDISABLE_BINUTILS=1 ..
$ make
stf_dump.cpp:108:139: error: no matching function for call to 'stf::disassemblers::MavisDisassembler::MavisDisassembler(std::string, stf::ISA, stf::INST_IEM, const bool&)'
  108 |         stf::Disassembler dis(findElfFromTrace(config.trace_filename), stf_reader.getISA(), stf_reader.getInitialIEM(), config.use_aliases);

note: candidate: 'stf::disassemblers::MavisDisassembler::MavisDisassembler(stf::ISA, stf::INST_IEM, bool)'
   34 |                 explicit MavisDisassembler(const ISA inst_set, const INST_IEM iem, const bool use_aliases)
```

**Reply by bdutro** [reply]
I have a separate fix for this I'll push soon.

**Reply by bdutro** [reply, marked as answer]
Fixed in https://github.com/sparcians/stf_tools/commit/a557070d622a03d1d110df38f743920dc0727452

**Reply by knute-mips** [reply]
Everything builds with mavis -- thanks!

**Comment by bdutro** [comment]
I've managed to reproduce this _once_ with the default Conda Bison (3.7.5), but after doing `rm -rf binutils-prefix` and trying again it works fine. There might be some kind of race condition in that version, because I was also able to build it with no errors using conda-forge Bison (3.8.2), the default Rocky Linux bison package (3.0.4), and the Arch Linux Bison (3.8.2).

**Reply by knute-mips** [reply]
Ok, thanks.  I'll try again when I get a free moment.  Thanks for the help!

**Reply by knute-mips** [reply]
Ok, got a free moment.  😄 

I was still having build issues (the same one) with Bison 3.8.2.  Even ran it by hand.  I backed Bison to 3.0.4 and now it works. 🤷‍♂️
